src/lineup/internal/cmds.ts

```diff
@@ -378,49 +389,98 @@ interface IRegExpFilter {
+}
+
+/**
+ * Similar to the `ILineUpStringFilter`, but the RegExp is replaced with `IRegExpFilter`
+ */
+interface ISerializableLineUpFilter {
+  /**
+   * Filter value
+   * Note that the RegExp is replaced with IRegExpFilter (compared to the `ILineUpStringFilter` interface)
+   */
+  filter: ILineUpStringFilterValue | IRegExpFilter;
+
+  /**
+   * Filter for missing values
+   */
+  filterMissing: boolean;
+}
+
+/**
+ *  Filter value
+ */
+type ILineUpStringFilterValue = string[] | string | null;
+
+/**
+ * Serializes LineUp string filter, which can contain RegExp objects to an IRegexFilter object.
+ * The return value of this function can be passed to `JSON.stringify()` and stored in the provenance graph.
  *
  * Background information:
  * The serialization step is necessary, because RegExp objects are converted into an empty object `{}` on `JSON.stringify`.
  * ```
  * JSON.stringify(/^123$/gm); // result: {}
  * ```
  *
- * @param value Input string or RegExp object
- * @returns {string | IRegExpFilter} Returns the input string or a plain `IRegExpFilter` object
+ * @internal
+ * @param filter LineUp filter object
+ * @returns Returns the `ISerializableLineUpFilter` object
  */
-function serializeRegExp(value: string | RegExp): string | IRegExpFilter {
-  if (!(value instanceof RegExp)) {
-    return value;
-  }
-  return {value: value.toString(), isRegExp: true};
+export function serializeLineUpFilter(filter: ILineUpStringFilter): ISerializableLineUpFilter {
+  const value = filter.filter;
+  const isRegexp = value instanceof RegExp;
+  return {
+    filter: {
+      value: isRegexp ? value.toString() : value as ILineUpStringFilterValue,
+      isRegExp: isRegexp
+    },
+    filterMissing: filter.filterMissing
+  };
 }
 
 /**
  * Restores a RegExp object from a given IRegExpFilter object.
  * In case a string is passed to this function no deserialization is applied.
- *
  * @param filter Filter as string or plain object matching the IRegExpFilter
- * @returns {string | RegExp| null} Returns the input string or the restored RegExp object
+ * @returns Returns the input string or the restored RegExp object
  */
-function restoreRegExp(filter: string | IRegExpFilter): string | RegExp {
-  if (filter === null || !(<IRegExpFilter>filter).isRegExp) {
-    return <string | null>filter;
+export function restoreRegExp(filter: ILineUpStringFilterValue | IRegExpFilter | ISerializableLineUpFilter, filterMissing = false): ILineUpStringFilter {
+  const isIRegExpFilter = ((filter: IRegExpFilter | ISerializableLineUpFilter): filter is IRegExpFilter => filter.hasOwnProperty('isRegExp'));
+  const isISerializableLineUpFilter = (filter: IRegExpFilter | ISerializableLineUpFilter): filter is ISerializableLineUpFilter => filter.hasOwnProperty('filterMissing');
+  if (filter === null || typeof filter === 'string' || Array.isArray(filter)) {
+    return {filter: <string | null>filter, filterMissing};
+  } else if (isIRegExpFilter(filter)) {
```

Comment:
Just a minor code style. I would add a new line before each else block

```suggestion

  } else if (isIRegExpFilter(filter)) {
```